# General params
show_gripper: True
max_plot_gripper: 10
table_extent: 0.15
gravity_accel: 9.81
custom_stable_poses: True

# Grasping params
gripper_dir: data/grippers
cache_dir: .dexnet

grasp_sampler: antipodal
target_num_grasps: 100
check_collisions: False
coll_check_num_grasp_rots: 32

friction_coef: 0.5
num_cone_faces: 8
max_grasp_sampling_iters: 5
grasp_samples_per_surface_point: 2
max_num_surface_points: 250
min_contact_dist: 0.01
dir_prior: 1.0

# Metric params
quality_scale: 0.3
metric_display_rate: 10

metrics:
  ferrari_canny:
    quality_method: ferrari_canny_L1
    friction_coef: 0.5
    num_cone_faces: 8
    soft_fingers: 1
    quality_type: quasi_static    

    # optional deterministic parameters
    torque_scaling: 0.01
    wrench_norm_thresh: 0.001
    wrench_regularizer: 0.0000000001
 
  force_closure:
    quality_method: force_closure
    friction_coef: 0.5
    num_cone_faces: 8
    soft_fingers: 1
    quality_type: quasi_static    

    # optional deterministic parameters
    torque_scaling: 0.01
    wrench_norm_thresh: 0.001
    wrench_regularizer: 0.0000000001

  robust_force_closure:
    # required parameters
    quality_method: force_closure
    friction_coef: 0.5
    num_cone_faces: 8
    soft_fingers: 1
    quality_type: robust_quasi_static
    
    # optional deterministic parameters
    torque_scaling: 0.01
    wrench_norm_thresh: 0.001
    wrench_regularizer: 0.0000000001
    
    # PARAMS for robust grasp metrics
    # uncertainty
    grasp_uncertainty:
      sigma_grasp_trans_x: 0.005
      sigma_grasp_trans_y: 0.005
      sigma_grasp_trans_z: 0.005
      sigma_grasp_rot_x: 0.001
      sigma_grasp_rot_y: 0.001
      sigma_grasp_rot_z: 0.001
      num_prealloc_samples: 0
    obj_uncertainty:
      sigma_obj_trans_x: 0.005
      sigma_obj_trans_y: 0.005
      sigma_obj_trans_z: 0.005
      sigma_obj_rot_x: 0.01
      sigma_obj_rot_y: 0.01
      sigma_obj_rot_z: 0.01
      sigma_obj_scale: 0.00
      num_prealloc_samples: 10
    params_uncertainty:
      sigma_friction_coef: 0.1
      num_prealloc_samples: 0

    # sampling params
    num_quality_samples: 50
    sampling_snapshot_rate: 10

  robust_partial_closure:
    # required parameters
    quality_method: partial_closure
    friction_coef: 0.5
    num_cone_faces: 8
    soft_fingers: 1
    quality_type: robust_quasi_static
    
    # optional deterministic parameters
    torque_scaling: 0.01
    wrench_norm_thresh: 0.001
    wrench_regularizer: 0.0000000001
    
    # PARAMS for robust grasp metrics
    # uncertainty
    grasp_uncertainty:
      sigma_grasp_trans_x: 0.005
      sigma_grasp_trans_y: 0.005
      sigma_grasp_trans_z: 0.005
      sigma_grasp_rot_x: 0.001
      sigma_grasp_rot_y: 0.001
      sigma_grasp_rot_z: 0.001
      num_prealloc_samples: 0
    obj_uncertainty:
      sigma_obj_trans_x: 0.005
      sigma_obj_trans_y: 0.005
      sigma_obj_trans_z: 0.005
      sigma_obj_rot_x: 0.01
      sigma_obj_rot_y: 0.01
      sigma_obj_rot_z: 0.01
      sigma_obj_scale: 0.00
      num_prealloc_samples: 10
    params_uncertainty:
      sigma_friction_coef: 0.1
      num_prealloc_samples: 0

    # sampling params
    num_quality_samples: 50
    sampling_snapshot_rate: 10

# Object gen params
# path_to_sdfgen: /home/jmahler/Libraries/SDFGen/bin/SDFGen
path_to_sdfgen: /home/cloudminds/SDFGen/bin/SDFGen
sdf_dim: 100
sdf_padding: 5
obj_scale: 1.0
mass_thresh: 0.01
stp_min_prob: 0.01
use_default_mass: False
default_mass: 1.0
obj_density: 4000.0